pkg/mcp/parser.go

```diff
@@ -279,12 +299,92 @@ func handleLoggingMethod(paramsMap map[string]interface{}) (string, map[string]i
 
 // handleCompletionMethod extracts resource ID for completion requests
 func handleCompletionMethod(paramsMap map[string]interface{}) (string, map[string]interface{}) {
+	// Check if ref is a map (PromptReference or ResourceTemplateReference)
+	if ref, ok := paramsMap["ref"].(map[string]interface{}); ok {
+		// Try to extract name for PromptReference
+		if name, ok := ref["name"].(string); ok {
+			return name, paramsMap
+		}
+		// Try to extract uri for ResourceTemplateReference
+		if uri, ok := ref["uri"].(string); ok {
+			return uri, paramsMap
+		}
+	}
+	// Fallback to string ref (legacy support)
 	if ref, ok := paramsMap["ref"].(string); ok {
-		return ref, nil
+		return ref, paramsMap
+	}
+	return "", paramsMap
+}
+
+// handleElicitationMethod extracts resource ID for elicitation requests
+func handleElicitationMethod(paramsMap map[string]interface{}) (string, map[string]interface{}) {
+	// The message field could be used as a resource identifier
+	if message, ok := paramsMap["message"].(string); ok {
+		return message, paramsMap
+	}
+	return "", paramsMap
+}
+
+// handleSamplingMethod extracts resource ID for sampling/createMessage requests
+func handleSamplingMethod(paramsMap map[string]interface{}) (string, map[string]interface{}) {
+	// Use model preferences or system prompt as identifier if available
+	if modelPrefs, ok := paramsMap["modelPreferences"].(map[string]interface{}); ok {
```

Comment:
we are not checking here about the param existing, being nil, etc... are we sure that it will always contains the expected values?